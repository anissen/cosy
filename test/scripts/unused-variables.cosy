// let a = 5 // not reported unused because its top-level
let b = 6
print b

{
    let c = 5 // c: unused variable
    let d = 6
    print d
}

fn x(arg1) {
    let local1 = 2 // local1: unused variable
    print arg1
}

fn y(arg2) { // arg2: unused variable
    let local2 = 2
    print local2
}

x(true)
y(false)

// ----------

fn f(_x) { // no error for unused argument
    let _y = 2 // no error for unused variable
    let _z = 5
    print _z // error for used variable starting with _
}

f('hello')
