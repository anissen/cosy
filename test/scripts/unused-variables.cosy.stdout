■ test/scripts/unused-variables.cosy, line 6:
4 | 
5 | {
6 |     let c = 5 // c: unused variable
            ^ Local variable is not used.
7 |     let d = 6
8 |     print d

■ test/scripts/unused-variables.cosy, line 12:
10 | 
11 | fn x(arg1) {
12 |     let local1 = 2 // local1: unused variable
             ^^^^^^ Local variable is not used.
13 |     print arg1
14 | }

■ test/scripts/unused-variables.cosy, line 16:
14 | }
15 | 
16 | fn y(arg2) { // arg2: unused variable
          ^^^^ Local variable is not used.
17 |     let local2 = 2
18 |     print local2

■ test/scripts/unused-variables.cosy, line 29:
27 |     let _y = 2 // no error for unused variable
28 |     let _z = 5
29 |     print _z // error for used variable starting with _
               ^^ Variables starting with _ are considered unused.
30 | }
31 | 

